tools/ibc-test/src/tests/ibc/transfer.rs

```diff
@@ -0,0 +1,162 @@
+    let client_a = {
+        let config_a = get_chain_config(relayer, chains.node_a.chain_id().value());
+        let url = config_a.websocket_addr.to_string();
+        Provider::connect(url).await?
+    };
+    let client_b = {
+        let config_b = get_chain_config(relayer, chains.node_b.chain_id().value());
+        let url = config_b.websocket_addr.to_string();
+        Provider::connect(url).await?
+    };
+    let denom = "AT".to_string();
+    let amount = 100;
+    let timeout_height = 0;
+    let source_port = channel.port_a.to_string();
+    let source_channel = channel.channel_id_a.to_string();
+    let (mock_transfer_address, ibc_handler_address) = {
+        let c = read_deployed_contracts(&chains.node_a.chain_driver().value().home_path)
+            .expect("failed to fetch deployed contracts");
+        (c.mock_transfer_contract_address, c.contract_address)
+    };
+    let (receiver, receiver_ibc_handler_address) = {
+        let c = read_deployed_contracts(&chains.node_b.chain_driver().value().home_path)
+            .expect("failed to fetch deployed contracts");
+        (c.mock_transfer_contract_address, c.contract_address)
+    };
+
+    let contract = new_mock_contract(
+        client_a.clone(),
+        relayer.registry.get_or_spawn(chains.chain_id_a().value())?,
+        mock_transfer_address,
+    )
+    .await?;
+    let a_chain_tip = client_a.get_block_number().await?;
+    let b_chain_tip = client_b.get_block_number().await?;
+
+    let tx = contract.send_transfer(
+        denom,
+        amount,
+        receiver,
+        source_port,
+        source_channel,
+        timeout_height,
+    );
+    let pending_tx = tx.send().await?;
+    pending_tx.await?;
+
+    // check send packet event
+    {
+        let filter = Filter::new()
+            .address(ibc_handler_address)
+            .from_block(a_chain_tip);
+        let logs = client_a.get_logs(&filter).await?;
+        assert_eq!(logs.len(), 1);
+
+        let event = OwnableIBCHandlerEvents::decode_log(&logs[0].clone().into())?;
+        log::info!("chain_a packet log: {event:?}");
+        assert!(matches!(
+            event,
+            OwnableIBCHandlerEvents::SendPacketFilter(..)
+        ));
+    }
+
+    // check receive packet event
+    let event = tokio::time::timeout(Duration::from_secs(30), async {
+        loop {
+            let filter = Filter::new()
+                .address(receiver_ibc_handler_address)
+                .from_block(b_chain_tip);
+            let logs = client_b.get_logs(&filter).await?;
+            for log in logs.into_iter() {
+                let event = OwnableIBCHandlerEvents::decode_log(&log.into()).unwrap();
+
+                if matches!(event, OwnableIBCHandlerEvents::ReceivePacketFilter(..)) {
+                    return eyre::Result::<_, ProviderError>::Ok(event);
+                }
+            }
+
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    })
+    .await??;
```

Comment:
this part assumes the counterparty is always Axon chain, and I wrote it in Js format is just testing relay function in two Axon chains, not considered the CKB

Reply:
After the CKB test is added we will call the ckb methods to retrive the events, for now we consider the counterparty is AXON because I don't now how to retrive packet events from CKB. What's your opinion?

Reply:
retrieving packet events from CKB, like CKB Endpoint's monitor does, is keeping to call cab-indexer until the expected packet cell found

Reply:
I see, but I think we can support it when the CKB test is added, it won't prevent current Axon tests.